### PR TITLE
Simplify client by removing tasks and channels

### PR DESCRIPTION
**Pull request by flxo**
The internal channels in the client for responses and notifications are not needed and make it impossible to report error conditions to the user of the lib. Remove the task in between the connection to the runtime and the client API.

### CODE REVIEW

File: northstar/src/api/codec.rs [hunk 1]
```diff
@@ -153,35 +154,35 @@ impl<T: Unpin + AsyncRead + AsyncWrite> futures::sink::Sink<model::Message> for
...
+
+//     fn mk_message() -> impl Strategy<Value = model::Message> {
+//         mk_simple_payload().prop_map(model::Message::Request(model::Request::Containers))
+//     }
+
+//     fn mk_simple_payload() -> impl Strategy<Value = model::Payload> {
+//         // TODO: Perhaps consider all the variants?
+//         Just(model::Payload::Request(model::Request::Containers))
+//     }
+// }
```
Comment (marcmo) [inline northstar/src/api/codec.rs:188]: should not be part of this commit. and should remain in, this is a good test

File: northstar/src/api/client.rs [hunk 2]
```diff
@@ -489,7 +484,39 @@ impl<'a> Client {
...
+    fn fuse(&mut self) {
+        self.fused = true;
+    }
+
+    /// Return Error::Stopped if the client is fused
+    fn fused(&self) -> Result<(), Error> {
+        if self.fused {
+            Err(Error::Stopped)
+        } else {
+            Ok(())
```
Comment (marcmo) [inline northstar/src/api/client.rs:519]: using the term `fuse` here is really confusing!

File: northstar/src/api/client.rs [hunk 3]
```diff
@@ -156,78 +155,16 @@ impl<'a> Client {
...
-
-        let mut connection = time::timeout(timeout, Self::connect(url, notifications, timeout))
+        let connection = time::timeout(timeout, Self::connect(url, notifications, timeout))
             .await
             .map_err(|_| Error::Timeout)??;
 
         debug!("Connected to {}", url);
 
-        task::spawn(async move {
-            loop {
```
Comment (marcmo) [inline northstar/src/api/client.rs:171]: removing the task indeed simplifies the code...nice

Comment (marcmo) [reply northstar/src/api/client.rs:171]: but how do notifications work when no request is pending?

File: northstar/src/api/codec.rs (hunk 1 above)
Comment (flxo) [reply northstar/src/api/codec.rs:188]: Ah. The proptest api confused me and I forgot the readd it. Basically this tests `serde`.

File: northstar/src/api/client.rs (hunk 2 above)
Comment (flxo) [reply northstar/src/api/client.rs:519]: It's borrowed from `futures`: https://docs.rs/futures/0.3.15/futures/future/trait.FutureExt.html#method.fuse.

File: northstar/src/api/client.rs (hunk 3 above)
Comment (flxo) [reply northstar/src/api/client.rs:171]: The `Stream` impl of `Client` checks the  notification buffer and if that's empty it polls the connection.
